prow/repoowners/endpoint_test.go

```diff
@@ -0,0 +1,154 @@
+			expectedCode: http.StatusOK,
+		},
+		{
+			name: "Missing content-type information",
+			body: generateQueryBody(&ownersRequest{
+				Org:        "org",
+				Repo:       "repo",
+				BaseCommit: "Master",
+				Paths:      []string{"src/dir", "src", "foo", "src/dir/conformance", "docs", "docs/file.md"},
+			}),
+			contentType:  "",
+			expectedBody: "",
+			expectedCode: http.StatusNotAcceptable,
+		},
+
+		{
+			name: "Empty paths",
+			body: generateQueryBody(&ownersRequest{
+				Org:        "org",
+				Repo:       "repo",
+				BaseCommit: "Master",
+				Paths:      nil,
+			}),
+			contentType:  "application/json",
+			expectedBody: "",
+			expectedCode: http.StatusBadRequest,
+		},
+		{
+			name: "Non-valid org",
+			body: generateQueryBody(&ownersRequest{
+				Org:        "9AA48AEB-30EB-4187-89E7-5FAEACC3EC91",
+				Repo:       "repo",
+				BaseCommit: "Master",
+				Paths:      []string{"src/dir", "src", "foo", "src/dir/conformance", "docs", "docs/file.md"},
+			}),
+			contentType:  "application/json",
+			expectedBody: "",
+			expectedCode: http.StatusBadRequest,
+		},
+		{
+			name: "Non-valid repo",
+			body: generateQueryBody(&ownersRequest{
+				Org:        "org",
+				Repo:       "A1A1693C-082C-41B1-9943-3A89B08A57C0",
+				BaseCommit: "Master",
+				Paths:      []string{"src/dir", "src", "foo", "src/dir/conformance", "docs", "docs/file.md"},
+			}),
+			contentType:  "application/json",
+			expectedBody: "",
+			expectedCode: http.StatusBadRequest,
+		},
+		{
+			name: "Non-valid basecommit",
+			body: generateQueryBody(&ownersRequest{
+				Org:        "org",
+				Repo:       "repo",
+				BaseCommit: "13202CD5-3068-46D4-8C1D-46F544ED39EF",
+				Paths:      []string{"src/dir", "src", "foo", "src/dir/conformance", "docs", "docs/file.md"},
+			}),
+			contentType:  "application/json",
+			expectedBody: "",
+			expectedCode: http.StatusBadRequest,
+		},
+		{
+			name:         "Empty body",
+			body:         []byte("{}"),
+			contentType:  "application/json",
+			expectedBody: "",
+			expectedCode: http.StatusBadRequest,
+		},
+		{
+			name:         "Bad-formed body",
+			body:         []byte("{"),
+			contentType:  "application/json",
+			expectedBody: "",
+			expectedCode: http.StatusBadRequest,
+		},
+		{
+			name:         "Empty body",
+			body:         []byte(""),
+			contentType:  "application/json",
+			expectedBody: "",
+			expectedCode: http.StatusBadRequest,
+		},
+	}
+
+	client, cleanup, err := getTestClient(testFiles, true, false, true, nil, nil, nil)
+	ownersServer := &OwnersServer{
+		OwnersClient: client,
+	}
+
+	for _, test := range tests {
+		t.Logf("Running scenario %q", test.name)
+
+		if err != nil {
+			t.Fatalf("Error creating test client: %v.", err)
+		}
+		defer cleanup()
+
+		req, err := http.NewRequest("POST", "/", bytes.NewBuffer(test.body))
```

Comment:
Always test an `err` right after it is created. Only carry on it is `nil` as expected.